pipelines/utilities/junit_to_qase.py

```diff
@@ -0,0 +1,274 @@
+#!/usr/bin/python
+import os
+import sys
+from urllib.parse import urlencode
+from datetime import date
+import requests
+from bs4 import BeautifulSoup
+
+
+def collect_test_results(test_cases):
+
+    test_results = []
+
+    for test_case in test_cases:
+
+        test_result = {
+            "case": "",
+            "case_id": "",
+            "suite": "",
+            "suite_id": "",
+            "skipped": "",
+            "failure": ""
+        }
+
+        test_suite = test_case.get("classname")
+        test_result['suite'] = test_suite.split(".")[-1]
+
+        test_result['case'] = test_case.get("name")
+
+        skipped = test_case.findChildren("skipped", recursive=False)
+        if len(skipped) > 0:
+            test_result['skipped'] = skipped[0].get("message")
+
+        failure = test_case.findChildren("failure", recursive=False)
+        if len(failure) > 0:
+            test_result['failure'] = failure[0].get("message")
+            test_result['failure'] += '\n\n' + failure[0].get_text()
+
+        test_results.append(test_result)
+
+    return test_results
+
+
+def get_suite_id(suite_name):
+
+    print(f"getting suite {suite_name}")
+
+    query_string = urlencode({ "search": suite_name })
+    headers = {
+        "accept": "application/json",
+        "Token": qase_token
+    }
+    url = f"https://api.qase.io/v1/suite/{qase_project}?{query_string}"
+
+    resp = requests.get(url, headers=headers)
+    res = resp.json()
+    if res["result"]["count"] > 0:
+        id = res["result"]["entities"][0]["id"]
+        print(f"got suite {suite_name} with id {id}")
+        return id
+    else:
+        print(f"missed suite {suite_name}")
+        return None
+
+
+def add_missing_suite(suite_name):
```

Comment:
This will add the test suit based on the test execution, right? Have we considered that Pytest cases & few robot test cases are already added in Qase which is under certain test suit. `automation->e2e-Pytest` & `automation->e2e-robot`

Reply:
Yes, it takes into account the test cases that have been added to Qase. That's why it only adds "missing" items.